code/go/0chain.net/smartcontract/storagesc/models.go

```diff
@@ -713,6 +715,77 @@ type StorageAllocation struct {
 	Name string `json:"name"`
 }
 
+func (sa *StorageAllocation) addToWritePool(
+	txn *transaction.Transaction,
+	mintNewTokens bool,
+	balances cstate.StateContextI,
+) error {
+	if !mintNewTokens {
+		if err := stakepool.CheckClientBalance(txn, balances); err != nil {
+			return fmt.Errorf("client balance check failed: %v", err)
+		}
+	}
+
+	if mintNewTokens {
+		if err := balances.AddMint(&state.Mint{
+			Minter:     ADDRESS,
+			ToClientID: ADDRESS,
+			Amount:     txn.Value,
+		}); err != nil {
+			return fmt.Errorf("minting tokens for write pool: %v", err)
+		}
+	} else {
+		transfer := state.NewTransfer(txn.ClientID, txn.ToClientID, currency.Coin(txn.Value))
+		if err := balances.AddTransfer(transfer); err != nil {
+			return fmt.Errorf("adding transfer to allocation pool: %v", err)
+		}
+	}
+
+	var err error
+	sa.WritePool, err = currency.AddCoin(sa.WritePool, txn.Value)
```

Comment:
If error happens, the sa.WritePool will be set to 0, we'd better keep it unchanged in case the sa would be used somewhere even the AddCoin failed.